File: drug_interactions.js
document.addEventListener('DOMContentLoaded', function() {
    const drugForm = document.getElementById('drug-interaction-form');
    const drug1Input = document.getElementById('drug1');
    const drug2Input = document.getElementById('drug2');
    const resultContainer = document.getElementById('interaction-result');
    const drugChips = document.querySelectorAll('.drug-chip');
    const loadingSpinner = document.getElementById('loading-spinner');
    
    // Example drug click handler
    if (drugChips) {
        drugChips.forEach(chip => {
            chip.addEventListener('click', function() {
                const drugName = this.textContent.trim();
                
                // If drug1 is empty, fill it, otherwise fill drug2
                if (!drug1Input.value) {
                    drug1Input.value = drugName;
                } else if (!drug2Input.value) {
                    drug2Input.value = drugName;
                } else {
                    // If both are filled, replace drug2
                    drug2Input.value = drugName;
                }
            });
        });
    }
    
    // Form submission handler
    if (drugForm) {
        drugForm.addEventListener('submit', function(e) {
            e.preventDefault();
            
            const drug1 = drug1Input.value.trim();
            const drug2 = drug2Input.value.trim();
            
            if (!drug1 || !drug2) {
                showNotification('Please enter both drugs', 'warning');
                return;
            }
            
            // Show loading spinner
            if (loadingSpinner) {
                loadingSpinner.classList.remove('hidden');
            }
            if (resultContainer) {
                resultContainer.innerHTML = '';
            }
            
            // Create form data
            const formData = new FormData();
            formData.append('drug1', drug1);
            formData.append('drug2', drug2);
            
            // Send request to check interactions
            fetch('/check_interaction', {
                method: 'POST',
                body: formData
            })
            .then(response => response.json())
            .then(data => {
                if (loadingSpinner) {
                    loadingSpinner.classList.add('hidden');
                }
                
                displayInteractionResult(data);
            })
            .catch(error => {
                console.error('Error checking drug interaction:', error);
                if (loadingSpinner) {
                    loadingSpinner.classList.add('hidden');
                }
                showNotification('An error occurred while checking drug interactions', 'danger');
            });
        });
    }
    
    // Function to display interaction results
    function displayInteractionResult(data) {
        if (!resultContainer) return;
        
        resultContainer.innerHTML = '';
        resultContainer.classList.add('slide-in-right');
        
        if (data.error) {
            resultContainer.innerHTML = `
                <div class="alert alert-warning">
                    <i class="fas fa-exclamation-triangle"></i> ${data.error}
                </div>
            `;
            return;
        }
        
        if (data.interaction) {
            const severityClass = getSeverityClass(data.severity);
            
            resultContainer.innerHTML = `
                <h3 class="mb-2">Interaction Found</h3>
                <div class="alert ${severityClass} mb-2">
                    <strong>Severity:</strong> ${data.severity || 'Unknown'}
                </div>
                <div class="mb-2">
                    <strong>Drug A:</strong> ${data.drug1}
                </div>
                <div class="mb-2">
                    <strong>Drug B:</strong> ${data.drug2}
                </div>
                <div class="mb-2">
                    <strong>Description:</strong>
                    <p>${data.description || 'No detailed description available.'}</p>
                </div>
                <div>
                    <strong>Recommendation:</strong>
                    <p>${getRecommendationByInteraction(data.severity) || 'Consult your healthcare provider for guidance.'}</p>
                </div>
            `;
        } else {
            resultContainer.innerHTML = `
                <div class="alert alert-success">
                    <i class="fas fa-check-circle"></i> No known interaction found between ${data.drug1} and ${data.drug2}.
                </div>
                <p class="mt-2">Always consult with a healthcare professional about all medications you are taking.</p>
            `;
        }
    }
    
    // Helper function to get severity class
    function getSeverityClass(severity) {
        if (!severity) return 'alert-warning';
        
        severity = severity.toLowerCase();
        if (severity.includes('major') || severity.includes('severe')) {
            return 'alert-danger';
        } else if (severity.includes('moderate')) {
            return 'alert-warning';
        } else if (severity.includes('minor') || severity.includes('mild')) {
            return 'alert-success';
        } else {
            return 'alert-warning';
        }
    }
    
    // Helper function to get recommendation based on interaction severity
    function getRecommendationByInteraction(severity) {
        if (!severity) return null;
        
        severity = severity.toLowerCase();
        if (severity.includes('major') || severity.includes('severe')) {
            return 'Avoid using these medications together. Consult your healthcare provider immediately.';
        } else if (severity.includes('moderate')) {
            return 'Use these medications together with caution. Monitor for side effects and consult your healthcare provider.';
        } else if (severity.includes('minor') || severity.includes('mild')) {
            return 'The benefits of using these medications together may outweigh the risks. Monitor for any unusual symptoms.';
        } else {
            return 'Consult your healthcare provider for guidance regarding these medications.';
        }
    }
});

File: drug_interactions.py
import logging
import pandas as pd

# Configure logging
logging.basicConfig(
    level=logging.DEBUG,
    format='%(asctime)s [%(levelname)s] %(message)s',
    handlers=[logging.StreamHandler()]
)
logger = logging.getLogger(__name__)

def get_severity_description(level):
    """
    Get description for severity level
    
    Args:
        level (str): Severity level (Major, Moderate, Minor)
        
    Returns:
        tuple: (description, recommendations)
    """
    if level == "Major":
        return (
            "Major interactions are highly clinically significant. Avoid combinations; the risk of the interaction outweighs the benefit.",
            "Strongly consider changing medication therapy and be alert for severe adverse events."
        )
    elif level == "Moderate":
        return (
            "Moderate interactions have moderate clinical significance. Usually avoid combinations; use only under special circumstances.",
            "Consider therapy modification and monitor closely for adverse effects."
        )
    elif level == "Minor":
        return (
            "Minor interactions have minimal clinical significance. Minimize risk; assess risk and consider alternative medication.",
            "Consider monitoring therapy and observe for any adverse effects."
        )
    else:
        return (
            "Interaction severity is not well documented.",
            "Exercise caution and consult your healthcare provider."
        )

def find_interaction(df, drug1, drug2):
    """
    Find interaction between two drugs in the dataset
    
    Args:
        df (pandas.DataFrame): DataFrame containing drug interaction data
        drug1 (str): Name of first drug
        drug2 (str): Name of second drug
        
    Returns:
        dict: Dictionary containing interaction information
    """
    if df is None:
        logger.error("Drug database is not available")
        return {'error': 'Drug database is not available'}
    
    try:
        # Convert drug names to lowercase for case-insensitive comparison
        drug1_lower = drug1.lower()
        drug2_lower = drug2.lower()
        
        # Check if drugs exist in dataset
        drug1_exists = any(df['Drug_A'].str.lower() == drug1_lower) or any(df['Drug_B'].str.lower() == drug1_lower)
        drug2_exists = any(df['Drug_A'].str.lower() == drug2_lower) or any(df['Drug_B'].str.lower() == drug2_lower)
        
        if not drug1_exists:
            logger.warning(f"Drug '{drug1}' not found in database")
            return {'error': f"Drug '{drug1}' not found in our database"}
        
        if not drug2_exists:
            logger.warning(f"Drug '{drug2}' not found in database")
            return {'error': f"Drug '{drug2}' not found in our database"}
        
        # Search for interaction in both directions (drug1-drug2 and drug2-drug1)
        interaction1 = df[(df['Drug_A'].str.lower() == drug1_lower) & 
                          (df['Drug_B'].str.lower() == drug2_lower)]
        
        interaction2 = df[(df['Drug_A'].str.lower() == drug2_lower) & 
                          (df['Drug_B'].str.lower() == drug1_lower)]
        
        # Combine results
        interaction = pd.concat([interaction1, interaction2])
        
        # If interaction found
        if not interaction.empty:
            # Get the first interaction record
            first_interaction = interaction.iloc[0]
            
            # Get severity level
            severity = first_interaction.get('Level', 'Unknown')
            
            # Get description and recommendations
            severity_description, recommendations = get_severity_description(severity)
            
            result = {
                'drug1': drug1,
                'drug2': drug2,
                'interaction': True,
                'severity': severity,
                'description': severity_description,
                'recommendations': recommendations
            }
            
            logger.info(f"Found {severity} interaction between {drug1} and {drug2}")
            return result
        else:
            # No interaction found
            logger.info(f"No interaction found between {drug1} and {drug2}")
            return {
                'drug1': drug1,
                'drug2': drug2,
                'interaction': False,
                'message': f"No known interactions found between {drug1} and {drug2}."
            }
    
    except Exception as e:
        logger.error(f"Error finding interaction: {str(e)}")
        return {'error': f"An error occurred: {str(e)}"}

def get_severity_level(severity_text):
    """
    Normalize severity level text to a standard format
    
    Args:
        severity_text (str): The severity text from the dataset
        
    Returns:
        str: Normalized severity level (Major, Moderate, Minor, or Unknown)
    """
    if not severity_text or not isinstance(severity_text, str):
        return "Unknown"
    
    severity_lower = severity_text.lower()
    
    if any(term in severity_lower for term in ['major', 'severe', 'contraindicated']):
        return "Major"
    elif any(term in severity_lower for term in ['moderate', 'significant']):
        return "Moderate"
    elif any(term in severity_lower for term in ['minor', 'mild', 'minimal']):
        return "Minor"
    else:
        return "Unknown"